tools/clang/test/HLSLFileCheck/hlsl/functions/recursion/lib_struct_member_unconditional.hlsl

```diff
@@ -6,14 +6,14 @@
 // The SCCP pass replaces the recursive call with an undef value,
 // which is why validation fails with a non-obvious error.
 
-// CHECK: error: recursive functions are not allowed: entry function calls recursive function 'func'
+// CHECK: error: recursive functions are not allowed: export function calls recursive function 'func'
```

Comment:
I feel like this error could be improved.

Originally, it just checked the entry function, so saying "entry function" was sufficient.  But now, we should probably say the function name that calls the recursive function, and I'm not sure saying 'export' here is helpful (might be confusing since we use it for entry functions as well).  So this could be: `"... allowed: function '<name>' calls ..."`

Next, I wonder if function name (`func`) could use a fully-qualified name instead (`S::func` in this case).
Finally, it could be useful to place a "note:" diag on the recursive function (or even better, on the recursive call site, but that'll be more work).